apps/vs-code-designer/src/app/commands/createNewCodeProject/cloudToLocal.ts

```diff
@@ -18,17 +19,110 @@ import { ZipFileStep } from '../createNewProject/createProjectSteps/ZipFileStep'
 import { OpenFolderStepCodeProject } from './CodeProjectBase/OpenFolderStepCodeProject';
 import { SetLogicAppName } from './CodeProjectBase/SetLogicAppNameStep';
 import { setWorkspaceName } from './CodeProjectBase/SetWorkspaceName';
+import { extend } from '@microsoft/logic-apps-shared';
 import { AzureWizard } from '@microsoft/vscode-azext-utils';
 import type { IActionContext } from '@microsoft/vscode-azext-utils';
 import { latestGAVersion, OpenBehavior } from '@microsoft/vscode-extension-logic-apps';
-import type { ICreateFunctionOptions, IFunctionWizardContext, ProjectLanguage } from '@microsoft/vscode-extension-logic-apps';
+import { extractConnectionDetails, changeAuthTypeToRaw } from './cloudToLocalHelper';
+import { getParametersJson } from '../../utils/codeless/parameter';
+import { writeFormattedJson } from '../../utils/fs';
+import { saveConnectionReferences, getConnectionsJson, getConnectionsAndSettingsToUpdate } from '../../utils/codeless/connection';
+import { parameterizeConnection } from '../../utils/codeless/parameterizer';
+import type {
+  ICreateFunctionOptions,
+  IFunctionWizardContext,
+  ProjectLanguage,
+  ConnectionsData,
+} from '@microsoft/vscode-extension-logic-apps';
 import { window } from 'vscode';
 import * as path from 'path';
-import * as util from 'util';
 import AdmZip = require('adm-zip');
 import * as fs from 'fs';
+
+// Constants
 const openFolder = true;
 
+// Function to create an AdmZip instance
+function createAdmZipInstance(zipFilePath: string) {
+  return new AdmZip(zipFilePath);
+}
+
+// Function to get zip entries
+function getZipEntries(zipFilePath: string) {
+  const zip = createAdmZipInstance(zipFilePath);
+  return zip.getEntries();
+}
+
+async function getSettings(context: IActionContext, connections: any, workspacePath: any) {
+  const tenantId = '';
+  const parametersFileName = 'parameters.json';
+  const workflowManagementBaseUrl = 'https://management.azure.com/';
+  const settingsRecord: Record<string, string> = {};
+
+  console.log('getSettings: Start');
+  const connectionReferences = connections.managedApiConnections || {};
+  console.log('getSettings: Connection references obtained', connectionReferences);
+  const parameters = await getParametersJson(workspacePath);
+  const connectionsJson = await getConnectionsJson(workspacePath);
+
+  const connectionsData: ConnectionsData = JSON.parse(connectionsJson);
+  Object.keys(connectionsData).forEach(async (connectionType) => {
```

Comment:
NIT: I think the forEach doesn't really handle async code. In this case it might be doing what you expect to do but not all the cases does it. Can you update it?
Here is a small text on the reason https://gist.github.com/joeytwiddle/37d2085425c049629b80956d3c618971

Reply:
Fixed in new Pr